light/txpool.go

```diff
@@ -396,7 +397,9 @@ func (pool *TxPool) validateTx(ctx context.Context, tx *types.Transaction) error
 
 	// Should supply enough intrinsic gas
 	header := pool.chain.GetHeaderByHash(pool.head)
-	gas, err := core.IntrinsicGas(tx.Data(), tx.To() == nil, header, currentState, tx.FeeCurrency(), pool.istanbul)
+
+	gasForAlternativeCurrency := blockchain_parameters.GetIntrinsicGasForAlternativeFeeCurrency(header, currentState)
```

Comment:
We should probably apply the optimization here where we only call `blockchain_parameters.GetIntrinsicGasForAlternativeFeeCurrency` if the fee currency is non-nil. Static calls a extra expensive from light clients due to requiring multiple network round trips.